geniza/corpus/models.py

```diff
@@ -535,7 +535,14 @@ def shelfmark(self):
     @property
     def original_date(self):
         """Generate formatted display for the document's original/historical date"""
-        return " ".join(
+        # separate with comma if both date and calendar are present, else just return whichever is present
+        # TODO: remove conditional once validation is implemented, since one will never be present alone
+        separator = (
+            ", "
+            if self.doc_date_original and self.get_doc_date_calendar_display()
+            else ""
+        )
+        return separator.join(
```

Comment:
my preferred pattern for this : 

```suggestion
        # separate with comma if both date and calendar are present, else just return whichever is present
        # TODO: remove conditional once validation is implemented, since one will never be present alone
        return ", ".join([v for v in [self.doc_date_original, self.get_doc_date_calendar_display()] if v])

```

Reply:
Much cleaner, thanks! I knew you'd have an idea 😄